File: Algorithm/Python/167. Two Sum II - Input array is sorted.py
# -*- coding: utf-8 -*-
# @Author: WuLC
# @Date:   2016-08-29 06:46:36
# @Last modified by:   WuLC
# @Last Modified time: 2016-08-29 06:46:52
# @Email: liangchaowu5@gmail.com

# two pointers
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        left, right = 0, len(numbers) - 1
        while left < right:
            tmp = numbers[left] + numbers[right]
            if tmp > target:
                right -= 1
            elif tmp < target:
                left += 1
            else:
                return [left+1, right+1]

File: Algorithm/Java/167. Two Sum II - Input array is sorted.java
/**
* Author: WuLC
* Date:   2016-10-09 19:26:10
* Last modified by:   WuLC
* Last Modified time: 2016-10-09 19:26:40
* Email: liangchaowu5@gmail.com
*/

// two pointers, O(n)
public class Solution 
{
    public int[] twoSum(int[] numbers, int target) 
    {
        int left = 0 , right = numbers.length-1;
        while (left < right)
        {   
            int sum = numbers[left] + numbers[right];
            if ( sum == target)
                break;
            else if (sum > target)
                right --;
            else
                left ++;
        }
        return new int[] {left+1, right+1};
    }
}